code/ModelAdmin.php

```diff
@@ -202,6 +204,9 @@ function ($form) {
                 $this->extend('updateSearchForm', $form);
             }
         ));
+        // GridFieldPaginator has to be added after filter header for it to function correctly
+        $listField->getConfig()->removeComponentsByType(GridFieldPaginator::class);
+        $listField->getConfig()->addComponent(Injector::inst()->get(GridFieldPaginator::class));
```

Comment:
This ignores the `page_length` config for modeladmin. You should be using `create` for items that have member properties, and `get` for global services / singletons.